### PR TITLE
Hiding requests if command is hidden

### LINKED ISSUE #414: Hidden command
It would be very useful to have some way to prevent commands from being displayed in the main command listing for a system in the UI.

The use-case for this would be for commands that are purely for generating dynamic choices options.

It's related to #399 and #269, but not quite the same.

### PR DESCRIPTION
**Pull request by west270**
Added a hidden flag into the request model that gets set if the corresponding command is hidden when the request is created.

Dependent on PR beer-garden/brewtils#288
Fixes #414

### CODE REVIEW

**Comment by TheBurchLog** [comment]
@hazmat345 What is the command to see the query type for Mongo?

File: src/app/beer_garden/api/http/handlers/v1/request.py [hunk 1]
```diff
@@ -510,6 +512,9 @@ def _parse_datatables_parameters(self) -> dict:
         if not include_children:
             filter_params["has_parent"] = False
 
+        if not include_hidden:
```
[the remark this answers is not in the document]
Comment (hazmat345) [reply src/app/beer_garden/api/http/handlers/v1/request.py:515]: This works with older requests without that field.

**Comment by hazmat345** [comment]
Looking at the logs for the mongodb container and hitting refresh on the request index page will give you output like this:

> mongodb_1      | 2021-02-04T22:46:31.805+0000 I  COMMAND  [conn123] command beer_garden.request command: count { count: "request", query: { has_parent: false, hidden: { $ne: true } }, lsid: { id: UUID("8a5cecc1-22cd-4fcc-95e8-5bfce30dfebc") }, $db: "beer_garden", $readPreference: { mode: "primaryPreferred" } } planSummary: IXSCAN { hidden: 1, has_parent: 1, created_at: -1, command: 1 } keysExamined:608966 docsExamined:0 numYields:4758 queryHash:C55B9950 planCacheKey:0B5602A3 reslen:45 locks:{ ReplicationStateTransition: { acquireCount: { w: 4759 } }, Global: { acquireCount: { r: 4759 } }, Database: { acquireCount: { r: 4759 } }, Collection: { acquireCount: { r: 4759 } }, Mutex: { acquireCount: { r: 1 } } } storage:{} protocol:op_msg 546ms

The important part is the plan summary: `planSummary: IXSCAN { hidden: 1, has_parent: 1, created_at: -1, command: 1 }`. This query was able to be satisfied by using the index with those 4 fields (one of the new indexes). This is the "good" case. The "bad" case would be if the plan summary was `COLSCAN` - that means that mongo has to inspect every document to satisfy the query.

**Review by hazmat345 — APPROVED** [review]
UI changes look good. I added some indexes and told the hinting logic about them. I think there might be some redundant indexes now, but I'm not going to worry about it. They aren't really hurting anything.

I played around with the request index page with ~600k requests and everything seems like it takes a reasonable amount of time.

I'm never 100% confidant when it comes to this stuff, but I'm at least confidant we aren't making the mongo situation significantly worse. So I'm going to go ahead and merge this.

**Pull request merged**